3_train.smk

```diff
@@ -1,15 +1,52 @@
 
 configfile: "3_train/train_config.yaml"
 
-# Train model
+# Save config files to output folder
+rule save_config:
+    input:
+        process_config = "2_process/process_config.yaml",
+        train_config = "3_train/train_config.yaml"
+    output:
+        process_config = "3_train/out/{data_source}/{run_id}/process_config.yaml",
+        train_config = "3_train/out/{data_source}/{run_id}/train_config.yaml"
+    shell:
+        """
+        mkdir -p 3_train/out/{wildcards.data_source}/{wildcards.run_id}
+        cp {input.process_config} {output.process_config}
+        cp {input.train_config} {output.train_config}
+        """
+
+# Train a model and save weights and metadata
+# Example:
+# $ snakemake --cores all 3_train/out/mntoha/4/hidden_size-50_weights.pt
 rule train_model:
     input:
-        npz_filepath = "2_process/out/{data_source}/train.npz"
+        npz_filepath = "2_process/out/{data_source}/train.npz",
+        # Include configs as inputs in order to save them to the output folder
+        process_config = "3_train/out/{data_source}/{run_id}/process_config.yaml",
+        train_config = "3_train/out/{data_source}/{run_id}/train_config.yaml"
```

Comment:
Just to confirm that my understanding is correct -- these don't actually get passed to the `main` function in `3_train/src/train.py`, but including them here ensures that `snakemake` will build the `save_config` rule, correct?

Reply:
Yes, exactly!